### PR TITLE
Add a .editorconfig file

### PR DESCRIPTION
**Pull request by stonefix**
I add a .editorconfig file for the issue #681

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,327 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = crlf
```
Comment (sclassen) [inline .editorconfig:5]: please use unix line ending

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+indent_size = 4
+indent_style = space
+insert_final_newline = false
```
Comment (sclassen) [inline .editorconfig:8]: I prefere a final new line in the file

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_block_brace_style = end_of_line
+ij_java_block_comment_at_first_column = true
+ij_java_call_parameters_new_line_after_left_paren = false
+ij_java_call_parameters_right_paren_on_new_line = false
+ij_java_call_parameters_wrap = off
+ij_java_case_statement_on_separate_line = true
+ij_java_catch_on_new_line = false
+ij_java_class_annotation_wrap = split_into_lines
+ij_java_class_brace_style = end_of_line
+ij_java_class_count_to_use_import_on_demand = 5
```
Comment (sclassen) [inline .editorconfig:72]: make this 999

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_doc_enable_formatting = true
+ij_java_doc_enable_leading_asterisks = true
+ij_java_doc_indent_on_continuation = false
+ij_java_doc_keep_empty_lines = true
+ij_java_doc_keep_empty_parameter_tag = true
+ij_java_doc_keep_empty_return_tag = true
+ij_java_doc_keep_empty_throws_tag = true
+ij_java_doc_keep_invalid_tags = true
+ij_java_doc_param_description_on_new_line = false
+ij_java_doc_preserve_line_breaks = false
```
Comment (sclassen) [inline .editorconfig:93]: please set this to true

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_doc_use_throws_not_exception_tag = true
+ij_java_else_on_new_line = false
+ij_java_enum_constants_wrap = off
+ij_java_extends_keyword_wrap = off
+ij_java_extends_list_wrap = off
+ij_java_field_annotation_wrap = split_into_lines
+ij_java_finally_on_new_line = false
+ij_java_for_brace_force = never
```
Comment (sclassen) [inline .editorconfig:101]: always braces please

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_for_statement_new_line_after_left_paren = false
+ij_java_for_statement_right_paren_on_new_line = false
+ij_java_for_statement_wrap = off
+ij_java_generate_final_locals = false
```
Comment (sclassen) [inline .editorconfig:105]: please use finals whenever possible

File: .editorconfig [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_generate_final_parameters = false
```
Comment (sclassen) [inline .editorconfig:106]: also true

File: .editorconfig [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_if_brace_force = never
```
Comment (sclassen) [inline .editorconfig:107]: also braces

File: .editorconfig [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_line_comment_add_space = false
+ij_java_line_comment_at_first_column = true
+ij_java_method_annotation_wrap = split_into_lines
+ij_java_method_brace_style = end_of_line
+ij_java_method_call_chain_wrap = off
+ij_java_method_parameters_new_line_after_left_paren = false
+ij_java_method_parameters_right_paren_on_new_line = false
+ij_java_method_parameters_wrap = off
+ij_java_modifier_list_wrap = false
+ij_java_names_count_to_use_import_on_demand = 3
```
Comment (sclassen) [inline .editorconfig:138]: also 999

File: .editorconfig [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_new_line_after_lparen_in_record_header = false
+ij_java_packages_to_use_import_on_demand = java.awt.*,javax.swing.*
```
Comment (sclassen) [inline .editorconfig:140]: no import on demand

File: .editorconfig [hunk 11]
```diff
@@ -0,0 +1,327 @@
...
+ij_java_visibility = public
+ij_java_while_brace_force = never
+ij_java_while_on_new_line = false
+ij_java_wrap_comments = false
+ij_java_wrap_first_method_in_call_chain = false
+ij_java_wrap_long_lines = false
+
+[*.properties]
+ij_properties_align_group_field_declarations = false
+ij_properties_keep_blank_lines = false
```
Comment (sclassen) [inline .editorconfig:251]: true

**Comment by stonefix** [comment]
Ok, thanks. I changed this after review

**Comment by hendrikebbers** [comment]
LGTM

**Pull request merged**